rotkehlchen/chain/ethereum/interfaces/balance.py

```diff
@@ -0,0 +1,51 @@
+from collections import defaultdict
+from typing import TYPE_CHECKING
+
+from rotkehlchen.accounting.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.assets.asset import Asset
+from rotkehlchen.chain.evm.types import string_to_evm_address
+from rotkehlchen.db.dbhandler import DBHandler
+from rotkehlchen.db.filtering import EvmEventFilterQuery
+from rotkehlchen.db.history_events import DBHistoryEvents
+from rotkehlchen.types import ChecksumEvmAddress
+
+if TYPE_CHECKING:
+    from rotkehlchen.accounting.structures.evm_event import EvmProduct
+
+
+class ProtocolWithBalance:
+
+    def __init__(self, counterparty: str, database: 'DBHandler'):
+        self.counterparty = counterparty
+        self.event_db = DBHistoryEvents(database)
+
+    def addresses_with_deposits(
```

Comment:
Hmmm I am not so sure if this needs to be in a class if it is the only function here. Maybe it can just be a standalone function?

Reply:
I added the other method of the interface. Having here an interface is good for future protocols that we can add imo

Reply:
Yes, makes sense